### DISCUSSION TITLE
Reset user password using admin account programatically or auth as a user using admin credentials?

### DISCUSSION THREAD

**Discussion by popovidis** [pocketbase/pocketbase#5007] [Q&A] [answered]
Hi, I am trying to either programatically reset the password of a user or auth as the user without a password using the admin credentials, is this possible with the js-sdk?

I am trying to somehow get magic links to work and I want to either reset the password to be the key to the magic link or to have a field being the otp and then authing as the user.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure that I understand the last part and how it is related to your question.

In general if you are authenticated as admin you should be able to directly modify the user password without providing the old password:

```js
await pb.admins.authWithPassword("test@example.com", "1234567890")

// change the password of some user by its id
// (if you don't have an id you can send a separate getFirstListItem request)
await pb.collection("users").update("USER_ID", {
    password: "new_pass",
    passwordConfirm: "new_pass",
})
```

**Reply by popovidis** [reply]
This is exactly what I am looking for thank you.

**Reply by ganigeorgiev** [reply]
_edit: It seems that you've deleted your last comment. I hate when users do this, but anyway..._

I don't understand what do you mean. There are no changes to the admin apis in the recent releases. Make sure that the error you are seeing is actually from the request you are submitting.

Note that if you are using the same JS SDK instance for both admin and regular user auth then this is insecure because requests from different users will overwrite the same `pb.authStore`. For more details please make sure to read https://github.com/pocketbase/pocketbase/discussions/5313.

If you still not able to identify the culprit, please open a new Q&A discussion with more details about the used PocketBase executable and JS SDK versions and a code sample illustrating the problem.

**Reply by popovidis** [reply]
I did delete my last comment because I realize I made a mistake where I overrode the admin privilege. So it is my issue. Thank you for the continued support
